src/Sentry/TransactionTracer.cs

```diff
@@ -210,6 +215,19 @@ public TransactionTracer(IHub hub, ITransactionContext context)
         Description = context.Description;
         Status = context.Status;
         IsSampled = context.IsSampled;
+
+        if (idleTimeout != null)
```

Comment:
Instead of just grabbing it from the options, can we optionally pass the idle timeout to the tracer somehow? The idea is to have `some` transactions that can be idle and time out on their own. The primary use case for this feature right now would be auto-instrumentation for things like UI interaction with no properly defined end.

Right now, with the `IdleTimeout` set on the options, all transactions, even the ones created by users, turn out to be idle.

Reply:
> Instead of just grabbing it from the options, can we optionally pass the idle timeout to the tracer somehow? The idea is to have `some` transactions that can be idle and time out on their own. The primary use case for this feature right now would be auto-instrumentation for things like UI interaction with no properly defined end.
> 
> Right now, with the `IdleTimeout` set on the options, all transactions, even the ones created by users, turn out to be idle.

Makes sense. I have dropped the idea of the extension method due to [this comment](https://github.com/getsentry/sentry-dotnet/pull/2452#issuecomment-1611620938), which sounded as questioning the need for an extension method. So was [this comment](https://github.com/getsentry/sentry-dotnet/pull/2452#issuecomment-1612147758).

But I can see when a user would like to have a transaction with an idle timeout w/o necessarily having all transactions configured with an idle timeout.

Should this be done via [`HubExtensions`](https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry/HubExtensions.cs)? If yes, I think that could be a wrap-up for this PR? Plus, a sample, maybe.

Reply:
@bruno-garcia suggested we add these extensions as/when customers had need of them (so as not to create a bunch of stuff we're maintaining but nobody is using).

@bitsandfoxes would you see a use case for having different timeouts for specific transactions on the Unity side, for example? If the default timeout was 3s then I could imagine someone wanting to bump it higher for a particular operation they knew was likely to take longer (e.g. a slow. performing query in some analytics database)... but that's just a hypothetical from me (it's not a real use case from an actual customer).

Reply:
I don't see Unity needing anything additional timeouts.